src/main/java/org/jabref/logic/importer/fetcher/CitationRelationFetcher.java

```diff
@@ -0,0 +1,137 @@
+package org.jabref.logic.importer.fetcher;
+
+import java.io.*;
+import java.net.SocketException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.SimpleDoubleProperty;
+import kong.unirest.json.JSONArray;
+import kong.unirest.json.JSONException;
+
+import org.jabref.logic.importer.EntryBasedFetcher;
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.preferences.JabRefPreferences;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class to fetch for an articles citation relations on opencitations.net's API
+ */
+public class CitationRelationFetcher implements EntryBasedFetcher {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(org.jabref.logic.importer.fetcher.CitationRelationFetcher.class);
+    private SearchType searchType = null;
+    private static final String BASIC_URL = "https://opencitations.net/index/api/v1/metadata/";
+    private final DoubleProperty progress = new SimpleDoubleProperty(0);
+
+    /**
+     * Possible search methods
+     */
+    public enum SearchType {
+        CITING("reference"),
+        CITEDBY("citation");
+
+        public final String label;
+
+        SearchType(String label) {
+            this.label = label;
+        }
+    }
+
+    public CitationRelationFetcher(SearchType searchType) {
+        this.searchType = searchType;
+        progress.set(0);
+    }
+
+    /**
+     * Executes the search method associated with searchType
+     *
+     * @param entry Entry to search relations for
+     * @return List of BibEntries found
+     */
+    @Override
+    public List<BibEntry> performSearch(BibEntry entry) throws FetcherException {
+        String doi = entry.getField(StandardField.DOI).orElse("");
+        if (searchType != null) {
+            List<BibEntry> list = new ArrayList<>();
+            try {
+                LOGGER.info("Search: " + BASIC_URL + doi);
```

Comment:
Change the loglevel to debug and use parameters:  e.g. `LOGGER.debug("Search: { }", BASIC_URL+DOI)` this is more performant
INFO level should be used when you need to inform the user about something.